GVFS/GVFS.Common/Git/GitRepo.cs

```diff
@@ -232,6 +232,8 @@ private LooseBlobState GetLooseBlobStateAtPath(string blobPath, Action<Stream, l
 
         private LooseBlobState GetLooseBlobState(string blobSha, Action<Stream, long> writeAction, out long size)
         {
+            // Ensure SHA path is lowercase for case-sensitive filesystems
+            blobSha = blobSha.ToLower();

```

Comment:
@chrisd8088 have you seen problems with `GetLooseBlobState` on Linux?

Rather than paying the cost of `ToLower` every time this method is called can you make sure that the source of the data (e.g. `Sha1Id.ToString()`) is always lowercase if it's not already?

If this is not feasible, can we check `CaseSensitiveFileSystem` before calling `ToLower()`?

Reply:
I'm just doing the latter (checking `CaseSensitiveFileSystem` before `ToLower()`) for the moment, although I agree it would be nice to track down all sources of uppercased SHA1s.  With luck, it's always from `Sha1Id`, but I'm not 100% sure of that yet.

Reply:
Thanks! I'm marking this comment as resolved.